File: solutions/typescript/proverb/1/proverb.ts
export function proverb(...args: string[]) {
  let result = '';
  for (let i = 0; i < args.length; i++) {
    if (i === 0) {
      result += `For want of a ${args[i]} the ${args[i + 1]} was lost.\n`;
    } else if (i === args.length - 1) {
      result += `And all for the want of a ${args[0]}.`;
    } else {
      result += `For want of a ${args[i]} the ${args[i + 1]} was lost.\n`;
    }
  }
  return result
}

File: solutions/go/proverb/1/proverb.go
// This is a "stub" file.  It's a little start on your solution.
// It's not a complete solution though; you have to write some code.

// Package proverb should have a package comment that summarizes what it's about.
// https://golang.org/doc/effective_go.html#commentary
package proverb

import "fmt"

// Proverb should have a comment documenting it.
func Proverb(rhyme []string) []string {

	var response []string

	if len(rhyme) == 0 {
		return response
	}

	for i := 0; len(rhyme) > 1 && i < len(rhyme)-1; i++ {
		response = append(response, fmt.Sprintf("For want of a %s the %s was lost.", rhyme[i], rhyme[i+1]))

	}

	response = append(response, fmt.Sprintf("And all for the want of a %s.", rhyme[0]))

	return response
}
